apps/site/assets/ts/stop/__tests__/StopPageRedesignTest.tsx

```diff
@@ -56,27 +52,27 @@ describe("StopPageRedesign", () => {
   });
 
   it("shows Loading without stop or routes", () => {
-    jest.spyOn(useRoute, "useRoutesByStop").mockImplementation(() => {
-      return undefined;
-    });
-    jest.spyOn(useStop, "default").mockImplementation(() => {
-      return undefined;
-    });
+    jest
+      .spyOn(useRoute, "useRoutesByStop")
+      .mockReturnValue({ status: FetchStatus.Data, data: undefined });
+    jest
+      .spyOn(useStop, "default")
+      .mockReturnValue({ status: FetchStatus.Data, data: undefined });
 
     render(<StopPageRedesign stopId="123" />);
-    expect(screen.queryByText("Loading...")).toBeDefined();
+    expect(screen.getByText("Loading...")).toBeDefined();
   });
 
   it("shows Loading without alerts", () => {
     jest
       .spyOn(useAlerts, "useAlertsByStop")
-      .mockImplementation(() => undefined);
+      .mockReturnValue({ status: FetchStatus.Data, data: undefined });
     jest
       .spyOn(useAlerts, "useAlertsByRoute")
-      .mockImplementation(() => undefined);
+      .mockReturnValue({ status: FetchStatus.Data, data: undefined });
 
     render(<StopPageRedesign stopId="123" />);
-    expect(screen.queryByText("Loading...")).toBeDefined();
+    expect(screen.getByText("Loading...")).toBeDefined();
```

Comment:
This test was passing, but the loading state was not actually being triggered. 
This is because `queryBy...`[ returns null](https://testing-library.com/docs/react-testing-library/cheatsheet/#queries) if there is no match, but `toBeDefined()` is only checking if the result is undefined. so `expect(null).toBeDefined()` passes. Updating these two instances to use `getByText` instead which will throw an error if there is no match. 
Going to look for other instances of `queryBy...` in the stop page redesign tests and address in a separate PR or make a ticket to do so.

Reply:
Opened separate PR: https://github.com/mbta/dotcom/pull/1617